src/monattack.cpp

```diff
@@ -2265,6 +2265,74 @@ void mattack::tentacle(monster *z, int index)
     g->u.check_dead_state();
 }
 
+void mattack::pull_enemy(monster *z, int index)
+{
+    int t;
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 3
+            || !z->sees(*target, t)) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    std::vector<point> line = line_to( z->pos2(), target->pos2(), t );
```

Comment:
Don't use `pos2()` here. This function will be going away soon. Use `pos()` and the vector of tripoints you get from it. It requires 2 `t` arguments, but it's fine if the second one is 0.